File: src/main/scala/com/ScalaLR/scala/myIterator.scala
package com.ScalaLR.scala

import com.JavaLr.java.Action

class myIterator[E <: Comparable[E] ](private val list: myList[E]) {

  def forEach(a: Action[E]): Unit = {
    var temp: myNode[E] = list.getHead
    if (temp != null) {
      for (i <- 0 until list.getSize) {
        a.toDo(temp.data)
        temp = temp.next
      }
    }
  }
}

File: src/main/Java/com/JavaLr/java/myIterator.java
package com.JavaLr.java;

public class myIterator<E extends Comparable<E>> {

    private myList<E> list = new myList<E>();

    public myIterator(myList<E> list){
        this.list = list;
    }

    public void forEch(Action<E> a){
        myNode<E> temp = list.getHead();
        if(temp != null)
            for(int i = 0; i < list.getSize();i++){
                a.toDo(temp.data);
                temp = temp.next;
            }
    }

}
